### PR TITLE
feat: update records

### PR DESCRIPTION
**Pull request by amordzon**
## Description
Do not delete all data but update them.

### CODE REVIEW

File: src/Schedulers/news.scheduler.ts
Comment (jzapiorkowski) [inline src/Schedulers/news.scheduler.ts:1]: create service that will update and create documents in news collection and use it in the scheduler. We may use manual scraping & updating in the future

**Review by Kusmierek — APPROVED** [review]
lgtm

File: src/Services/UpdateRecords/enumerateData.service.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+type UpdateOrCreateFunction<RecordType, CompareType> = (
+    comparisonObject: any,
```
Comment (Kusmierek) [inline src/Services/UpdateRecords/enumerateData.service.ts:2]: Don't use any, if it's a generic function then use T

File: src/Services/UpdateRecords/enumerateData.service.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    updateOrCreateData: UpdateOrCreateFunction<RecordType, CompareType>,
+    generateComparisonObject: (record: RecordType) => any
+) => {
+    if (records && records.length > 0) {
+        records.forEach(async (record: RecordType) => {
+            const comparisonObject = generateComparisonObject(record);
+            await updateOrCreateData(comparisonObject, record, compareValues);
+        });
+    }
+};
```
Comment (Kusmierek) [inline src/Services/UpdateRecords/enumerateData.service.ts:19]:
```suggestion
export const enumerateData = <RecordType, CompareType>(
    records: RecordType[],
    compareValues: CompareType,
    updateOrCreateData: UpdateOrCreateFunction<RecordType, CompareType>,
    comparisionKey: key of RecordType
) => {
    if (records && records.length > 0) {
        records.forEach(async (record: RecordType) => {
            const comparisonObject = record[comparisionKey]
            await updateOrCreateData(comparisonObject, record, compareValues);
        });
    }
};
```

File: src/Services/UpdateRecords/enumerateData.service.ts (hunk 1 above)
Comment (Kusmierek) [inline src/Services/UpdateRecords/enumerateData.service.ts:2]: any

File: src/Services/UpdateRecords/enumerateData.service.ts (hunk 2 above)
Comment (Kusmierek) [inline src/Services/UpdateRecords/enumerateData.service.ts:19]: General is good practice in some tricky logic, to add comments about params. Here it's how it looks
image
With such solution, If You hover on function in IDE then already you can already get a information about params and what those params do. It makes development faster for other coworkers 🐫

**Review by Kusmierek — COMMENTED** [review]
nice function, the tough one

File: src/Services/UpdateRecords/enumerateData.service.ts (hunk 2 above)
Comment (amordzon) [reply src/Services/UpdateRecords/enumerateData.service.ts:19]: not sure if this is a better way in this case

**Pull request merged**